File: app-components/src/main/java/org/synsystems/onlypass/components/cryptography/random/SecureRandomModule.java
package org.synsystems.onlypass.components.cryptography.random;

import org.synsystems.onlypass.components.AppScope;

import java.security.SecureRandom;

import dagger.Module;
import dagger.Provides;

@Module
@AppScope
public class SecureRandomModule {
  @Provides
  @AppScope
  public SecureRandom provideSecureRandom() {
    return new SecureRandom();
  }
}

File: app-components/src/main/java/org/synsystems/onlypass/components/cryptography/random/SecureRandomModule.kt
package org.synsystems.onlypass.components.cryptography.random

import dagger.Module
import dagger.Provides
import org.synsystems.onlypass.components.AppScope
import java.security.SecureRandom

@Module
class SecureRandomModule {

  @Provides
  @AppScope
  fun provideSecureRandom() = SecureRandom()
}
